src/formats/internal/memreader.c

```diff
@@ -5,6 +5,11 @@
 #include "formats/internal/memreader.h"
 #include "formats/internal/reader.h"
 #include "utils/allocator.h"
+#ifdef N64_BUILD
+#include <libdragon.h>
+#elif BIG_ENDIAN_BUILD
+#define BITCAST_I2F(x) (float)(x)
+#endif
```

Comment:
This is not a correct fallback implementation of [libdragon's BITCAST_I2F](https://libdragon.dev/ref/fmath_8h_source.html).
BITCAST_I2F is more similar to (C++) `bit_cast<float>(x)`.

Perhaps it would be easier to use memcpy directly and avoid the utility function?
```c

#ifdef BIG_ENDIAN_BUILD
    uint32_t fl;
    fl = *((uint32_t *)(reader->buf + reader->pos));
    fl = __builtin_bswap32(fl);
    memcpy(&r, &fl, sizeof(fl));
    reader->pos += sizeof(fl);
#else
```